### ISSUE TITLE
expt 0.4.1 breaks in python 3.10

### ISSUE THREAD

**Issue by vwxyzjn** [wookayin/expt#24]
```
import expt
expt.Experiment("Comparison")
```
seems to break in python 3.10.

<img width="1094" alt="image" src="https://github.com/wookayin/expt/assets/5555347/cf9a8024-e1b1-4c75-8306-461f0375ae23">

**Comment by wookayin** [comment]
Thanks for the report. Yes, indeed the line
```
      hypotheses: Iterable[Hypothesis] = None,
```
is a not typing.

This is no longer a problem in the master branch as have been fixed since 1166a49f8c18fd7a56cab25f76ead00b3d39b10f, but I will backport the fix because people would want a stable version that works.

A workaround would be specifying the argument explicitly: `Experiment(name, hypotheses=())`

**Comment by wookayin** [comment]
This is actually a `typeguard` issue rather than python 3.10. I've released v0.4.2 which contains a backport patch.
